src/Digdir.Domain.Dialogporten.Infrastructure/Altinn/Authorization/AltinnAuthorizationClient.cs

```diff
@@ -71,7 +79,23 @@ internal sealed class AltinnAuthorizationClient : IAltinnAuthorization
             => await PerformNonScalableDialogSearchAuthorization(request, token), token: cancellationToken);
     }
 
-    private async Task<DialogSearchAuthorizationResult> PerformNonScalableDialogSearchAuthorization(DialogSearchAuthorizationRequest request, CancellationToken cancellationToken)
+    public async Task<AuthorizedPartiesResult> GetAuthorizedParties(IPartyIdentifier authenticatedParty,
+        CancellationToken cancellationToken = default)
+    {
+        var authorizedPartiesRequest = new AuthorizedPartiesRequest(authenticatedParty);
+        return await _cache.GetOrSetAsync(authorizedPartiesRequest.GenerateCacheKey(), async token
+            => await PerformAuthorizedPartiesRequest(authorizedPartiesRequest, token), token: cancellationToken);
+    }
```

Comment:
Good catch. Same problem with the PDP API call, as well as all the other clients, I suppose, so we have some fixing to do.

From how I understand the [fail safe mechanism](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/FailSafe.md#-a-practical-example) in FusionCache, we should throw an exception if the factory for some reason is unable to produce a valid value (server down, non-successful response, unable to deserialize). This will be caught by FusionCache, and the stale value (if it exists) will be reinserted into the cache for `FailSafeThrottleDuration` units of time, until which a new attempts will be made until the factory works again or `FailSafeMaxDuration` is reached. I'm guessing that if there isn't any stale value to begin with, the exception will bubble up.

Reply:
>the exception will bubble up,

In the case of authorized party endpoint, we would like to return a "valid" but empty result to the client right? Or would that be unexpected in this case too? 

As a general rule then:
- If it's an unexpected response, throw error and let the fail safe mechanism do it's thing
- If it's sort of expected, but we wouldn't like to cache the result with the default duration because we expect data to change in the near future, we could rather use [Adaptive Caching](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/AdaptiveCaching.md) like in this case. Ensuring that the client receives valid data, but opens up for polling/realtime or refetching of data from the client without explosions or empty data. We use a lot of third party systems and it might be useful to gracefully handle some of the cases, although we want to indicate to the client in most cases that something is not as expected 🤔

It might be that the first point would cover most cases though? A combination of polly + fail safe

Reply:
> In the case of authorized party endpoint, we would like to return a "valid" but empty result to the client right? Or would that be unexpected in this case too?

I don't think returning an empty list of authorized parties would make sense. It would at the very least require the client (ie. "arbeidsflate") to build special logic for this case where it has to make assumptions ("the list of authorized parties came back empty, but I guess the user can always represent him/self") in order to move forward in any useful sense, and moreover incorporate the same adaptivity in any of its own caches. There might also be other cases where an empty response might be perfectly normal, and thus impossible to tell apart from these cases where reponse is just transiently empty due to some (hopefully) shortlived error (which again should affect caching decisions). 

In any case, I think it's probably better to just let it blow up and tell the user that "we dun goofed; try again later", instead of trying to gloss it over and eg. end up displaying an empty/partial inbox, which will most likely cause confusion (and support requests).

Reply:
That said, using Polly with retries with exponential back-off should be the norm for all these HTTP requests, and combined with the option to temporarily re-use stale data via the FailSafe mechanism ought to give the robustness we need. (By the way, we need to have the hard timeouts for factory methods to take into account retries/backoff)

Reply:
Suggestion, merge this as is and create a general issue for this? The empty result should be an edge case here anyways?

Reply:
> Suggestion, merge as is and create a general issue for this? The empty result should be an edge case here anyways?

Sounds good.